### PR TITLE
Undocumented/calendar event modal

### PR DESCRIPTION
**Pull request by helihard**
## Description
This PR re-organises components in the calendar event modal and provides a new visualisation of signups and booked participants.

## Changes
* Lift up "Date & Time" and "Location" directly below the event selection checkbox, event title and status dot
* Display "Signups", "Booked" and "Notified" vertically, and in that order
* Hide the "Signups" title if no one has signed up to the event
* Display up to 17 avatars of signups, and display any surplus as a chip
* Display up to 17 avatars of booked participants, and display any surplus as a chip
* Place the "Go to event page ->" button in horisontal line with the ellipsis menu if the event has been published; otherwise, display the "Go to event page ->" button on its own line, and below that the "Publish" button in horisontal line with the ellipsis menu

## Notes to reviewer
None

### CODE REVIEW

File: src/features/events/l10n/messageIds.ts [hunk 1]
```diff
@@ -125,6 +125,12 @@ export default makeMessages('feat.events', {
     reminded: m('Reminded'),
     shifts: m('Shifts'),
     signups: m('Signups'),
+    surplusAvailable: m<{ numOthers: number }>(
+      '{numOthers, plural, =1 { + 1 more participant} other {+ # more participants}}'
+    ),
+    surplusSigned: m<{ numOthers: number }>(
+      '{numOthers, plural, =1 { + 1 more signed up} other {+ # more signed up}}'
+    ),
```
Comment (richardolsson) [inline src/features/events/l10n/messageIds.ts:133]: Are these used anymore?

File: src/features/events/components/EventPopper/SingleEvent.tsx [hunk 2]
```diff
@@ -272,38 +294,57 @@ const SingleEvent: FC<SingleEventProps> = ({ event, onClickAway }) => {
...
       )}
-      <Box
-        alignItems="center"
-        display="flex"
-        justifyContent="flex-end"
-        marginBottom={2}
-      >
-        <NextLink href={getEventUrl(event)} legacyBehavior passHref>
-          <Link underline="none">
+      {showPublishButton ? (
```
Comment (richardolsson) [inline src/features/events/components/EventPopper/SingleEvent.tsx:297]: I think this ternary should be replaced with two separate expressions on the pattern `{showPublishButton && (<></>)}` instead. In a long vertical chunk of code, I find it very difficult to find the place where the ternary "switches" from the expr-if-true to expr-if-false.

File: src/features/events/components/EventPopper/ParticipantAvatars.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  participants: ZetkinEventParticipant[] | ZetkinEventResponse[];
+}
+
+const ParticipantAvatars = ({
+  participants: list,
+  orgId,
+}: ParticipantAvatarsProps) => {
+  return (
+    <Box alignItems="center" display="flex" flexWrap="wrap" gap={0.5}>
+      <AvatarGroup
```
Comment (richardolsson) [inline src/features/events/components/EventPopper/ParticipantAvatars.tsx:22]: I had no idea about this component! I think it's great that you were able to find a component that takes care of the capping for us. 💯

File: src/features/events/components/EventPopper/ParticipantAvatars.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        max={18}
+        sx={{
+          /* surplus chip styling */
```
Comment (richardolsson) [inline src/features/events/components/EventPopper/ParticipantAvatars.tsx:25]: There's something about the look of this chip to me that feels a bit off. I think it's the lack of padding on the left and right?

image

**Review by richardolsson — CHANGES_REQUESTED** [review]
Nice work! I have looked at the preview build and at the code. I'm sure a lot of organizers will be happy about this!

I found some simple but (IMO) important things that I've commented on below.

I also wanted to point out something for the purpose of learning, but which requires no changes to the code. I don't believe "modal" (as you've called the branch and the title of this PR) is the proper term for this element, that we have instead called the "popper". Although I'm aware that "modal" is sometimes sloppily used to describe any UI that opens on top of another UI, the [defining characteristic of a modal is that it blocks](https://en.wikipedia.org/wiki/Modal_window#:~:text=a%20mode%20that%20disables%20user%20interaction%20with%20the%20main%20window%20but%20keeps%20it%20visible%2C%20with%20the%20modal%20window%20as%20a%20child%20window%20in%20front%20of%20it.%20Users%20must%20interact%20with%20the%20modal%20window%20before%20they%20can%20return%20to%20the%20parent%20window) interaction with the underlying UI, which this popper does not.

File: src/features/events/l10n/messageIds.ts (hunk 1 above)
Comment (helihard) [reply src/features/events/l10n/messageIds.ts:133]: No, will remove them. Name of element noted! It was called CalendarEventModal in the Figma sketch so I just went with that

**Review by ziggabyte — APPROVED** [review]
I'm giving you the green light to merge this! :D

**Pull request merged**